src/room/CallEventAudioRenderer.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2024 New Vector Ltd.
+
+SPDX-License-Identifier: AGPL-3.0-only
+Please see LICENSE in the repository root for full details.
+*/
+
+import { ReactNode, useDeferredValue, useEffect, useRef } from "react";
+import { useObservableEagerState } from "observable-hooks";
+
+import {
+  playReactionsSound,
+  soundEffectVolumeSetting as effectSoundVolumeSetting,
+  useSetting,
+} from "../settings/settings";
+import { CallViewModel } from "../state/CallViewModel";
+import enterCallSoundMp3 from "../sound/join_call.mp3";
+import enterCallSoundOgg from "../sound/join_call.ogg";
+import leftCallSoundMp3 from "../sound/left_call.mp3";
+import leftCallSoundOgg from "../sound/left_call.ogg";
+
+// Do not play any sounds if the participant count has exceeded this
+// number.
+export const MAX_PARTICIPANT_COUNT_FOR_SOUND = 8;
+
+export function CallEventAudioRenderer({
+  vm,
+}: {
+  vm: CallViewModel;
+}): ReactNode {
+  const [shouldPlay] = useSetting(playReactionsSound);
+  const [effectSoundVolume] = useSetting(effectSoundVolumeSetting);
+  const memberIds = useObservableEagerState(vm.userMediaIds);
+  const previousMembers = useDeferredValue(memberIds);
+  const callEntered = useRef<HTMLAudioElement>(null);
+  const callLeft = useRef<HTMLAudioElement>(null);
+
+  useEffect(() => {
+    if (memberIds.length > MAX_PARTICIPANT_COUNT_FOR_SOUND) {
+      return;
+    }
+
+    const memberLeft = !!previousMembers.filter((m) => !memberIds.includes(m))
+      .length;
+    const memberJoined = !!memberIds.filter((m) => !previousMembers.includes(m))
+      .length;
+
+    if (callEntered.current && callEntered.current?.paused && memberJoined) {
+      callEntered.current.volume = effectSoundVolume;
+      void callEntered.current.play();
+    }
```

Comment:
Due to the magic of <audio />, we are currently limited to only playing the sound when it's not already playing so this isn't actually an issue at present.

However, I still think it's probably worth doing this for performance.

Reply:
In this case I am almost inclined to see what a reasonable amount of parallel joins is (maybe 2-3) and create 2 or 3 players so that we can do the logic to only drop joines once all of them are already in playing state?

Reply:
nice